src/core/lib/iomgr/tcp_posix.cc

```diff
@@ -732,10 +730,14 @@ static bool process_errors(grpc_tcp* tcp) {
 
     if (msg.msg_controllen == 0) {
       /* There was no control message found. It was probably spurious. */
-      return true;
+      return;
+    }
+    auto cmsg = CMSG_FIRSTHDR(&msg);
+    if (cmsg == nullptr || cmsg->cmsg_len == 0) {
+      /* No control message found. */
+      return;
     }
-    for (auto cmsg = CMSG_FIRSTHDR(&msg); cmsg && cmsg->cmsg_len;
```

Comment:
instead of adding that if and the do-while, I would suggest just adding a boolean or a count and return if it's not set. The number of branches would be identical but it's easier to ready IMHO. WDYT?
```
seen = false
for (....) { 
   seen = true
}
if (!seen) return
```

Reply:
yes! I'll do that. Thanks!

Reply:
Done!